controllers/kserve_inferenceservice_controller_test.go

```diff
@@ -95,6 +103,89 @@ var _ = Describe("The Openshift Kserve model controller", func() {
 			}, timeout, interval).Should(Succeed())
 		})
 
+		It("With a new Kserve InferenceService, serving cert annotation should be added to the runtime Service object.", func() {
+			// Create a new InferenceService
+			inferenceService := &kservev1beta1.InferenceService{}
+			err := convertToStructuredResource(KserveInferenceServicePath1, inferenceService)
+			Expect(err).NotTo(HaveOccurred())
+			inferenceService.SetNamespace(testNs)
+			Expect(cli.Create(ctx, inferenceService)).Should(Succeed())
+
+			// Stub: Create a Runtime Service, which must be created by the OpenDataHub operator.
+			svc := &corev1.Service{}
+			err = convertToStructuredResource(testIsvcSvcPath, svc)
+			Expect(err).NotTo(HaveOccurred())
+			svc.SetNamespace(inferenceService.Namespace)
+			Expect(cli.Create(ctx, svc)).Should(Succeed())
+
+			// Update the URL of the InferenceService to indicate it is ready.
+			deployedInferenceService := &kservev1beta1.InferenceService{}
+			err = cli.Get(ctx, types.NamespacedName{Name: inferenceService.Name, Namespace: inferenceService.Namespace}, deployedInferenceService)
+			Expect(err).NotTo(HaveOccurred())
+
+			url, err := apis.ParseURL("https://example-onnx-mnist-default.test.com")
+			Expect(err).NotTo(HaveOccurred())
+			deployedInferenceService.Status.URL = url
+
+			err = cli.Status().Update(ctx, deployedInferenceService)
+			Expect(err).NotTo(HaveOccurred())
+
+			isvcService, err := waitForService(cli, testNs, inferenceService.Name, 5, 2*time.Second)
```

Comment:
Better to use `Eventually` and let the test framework to re-try for yourself.

Reply:
is it using async way? this test needs to proceed sync way. like if there is not the Service, it would not move forward to next line. `Eventually` will support this?

Reply:
Yes, `Eventually` blocks the program flow until it times out, or its assertion succeeds.

Reply:
updated